tests/utils.go

```diff
@@ -2816,40 +2818,72 @@ func WaitForVMICondition(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachi
 	}, time.Duration(timeoutSec)*time.Second, 2).Should(BeTrue(), fmt.Sprintf("Should have %s condition", conditionType))
 }
 
-func WaitForVMIConditionRemovedOrFalse(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance, conditionType v1.VirtualMachineInstanceConditionType, timeoutSec int) {
-	By(fmt.Sprintf("Waiting for %s condition removed or false", conditionType))
-	EventuallyWithOffset(1, func() bool {
-		updatedVmi, err := virtClient.VirtualMachineInstance(util2.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		for _, condition := range updatedVmi.Status.Conditions {
-			if condition.Type == conditionType && condition.Status == k8sv1.ConditionTrue {
-				return true
-			}
-		}
-		return false
-	}, time.Duration(timeoutSec)*time.Second, 2).Should(BeFalse(), fmt.Sprintf("Should have no or false %s condition", conditionType))
+func matcherGeneralCondition(typeMaper func(interface{}) string, conditions func(interface{}) interface{}, conditionType string, expected k8sv1.ConditionStatus, options gstruct.Options) types.GomegaMatcher {
+	options |= gstruct.IgnoreExtras
+	return WithTransform(conditions,
+		gstruct.MatchElements(typeMaper, options,
+			gstruct.Elements{
+				conditionType: gstruct.MatchFields(gstruct.IgnoreExtras,
+					gstruct.Fields{
+						"Status": Equal(expected),
+					}),
+			}))
 }
 
-func WaitForVMCondition(virtClient kubecli.KubevirtClient, vm *v1.VirtualMachine, conditionType v1.VirtualMachineConditionType, timeoutSec int) {
-	By(fmt.Sprintf("Waiting for %s condition", conditionType))
-	EventuallyWithOffset(1, func() bool {
-		updatedVm, err := virtClient.VirtualMachine(util2.NamespaceTestDefault).Get(vm.Name, &metav1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		for _, condition := range updatedVm.Status.Conditions {
-			if condition.Type == conditionType && condition.Status == k8sv1.ConditionTrue {
-				return true
-			}
-		}
-		return false
-	}, time.Duration(timeoutSec)*time.Second, 2).Should(BeTrue(), fmt.Sprintf("Should have %s condition", conditionType))
+func matcherVMICondition(conditionType v1.VirtualMachineInstanceConditionType, expected k8sv1.ConditionStatus, options gstruct.Options) types.GomegaMatcher {
+	conditions := func(obj interface{}) interface{} {
```

Comment:
@fossedihelm Your comment got me thinking that using unstructured here would probably generalize this even more...

Reply:
@xpivarc Yes it could be generalized with unstructured to act with VMI and VM both. Sorry to not having notice that.